### PR TITLE
feat: refact updateStatusConditionIfNotMatch and updateStatusNodeListIfNotMatch

### PR DESCRIPTION
**Pull request by bangqipropel**
**Reason for Change**:
refact updateStatusConditionIfNotMatch and updateStatusNodeListIfNotMatch

**Requirements**

- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

**Comment by zhuangqh** [comment]
please do `make manifests && make generate` before file your pr

File: pkg/controllers/ragengine_status_test.go [hunk 1]
```diff
@@ -110,7 +110,8 @@ func TestRAGEngineUpdateStatusConditionIfNotMatch(t *testing.T) {
 			},
 		}
 
-		err := reconciler.updateStatusConditionIfNotMatch(ctx, ragengine, conditionType, conditionStatus, conditionReason, conditionMessage)
+		err := updateStatusConditionIfNotMatch(ctx, ragengine, ragengine, reconciler.Client, &client.ObjectKey{Name: ragengine.Name, Namespace: ragengine.Namespace},
```
Comment (ishaansehgal99) [inline pkg/controllers/ragengine_status_test.go:113]: Same param passed twice?

File: pkg/controllers/ragengine_status_test.go [hunk 2]
```diff
@@ -138,7 +139,8 @@ func TestRAGEngineUpdateStatusConditionIfNotMatch(t *testing.T) {
 		mockClient.On("Get", mock.IsType(context.Background()), mock.Anything, mock.IsType(&kaitov1alpha1.RAGEngine{}), mock.Anything).Return(nil)
 		mockClient.StatusMock.On("Update", mock.IsType(context.Background()), mock.IsType(&kaitov1alpha1.RAGEngine{}), mock.Anything).Return(nil)
 
-		err := reconciler.updateStatusConditionIfNotMatch(ctx, ragengine, conditionType, conditionStatus, conditionReason, conditionMessage)
+		err := updateStatusConditionIfNotMatch(ctx, ragengine, ragengine, reconciler.Client, &client.ObjectKey{Name: ragengine.Name, Namespace: ragengine.Namespace},
+			ragengine.Status.Conditions, conditionType, conditionStatus, conditionReason, conditionMessage)
```
Comment (ishaansehgal99) [inline pkg/controllers/ragengine_status_test.go:143]: Same param passed twice?

**Comment by ishaansehgal99** [comment]
Already addressed

**Pull request closed without merging**